portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -56,56 +59,43 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         for (Entity entity : results.asIterable()) {
             String comment = (String) entity.getProperty("text");
             String sender = (String) entity.getProperty("sender");
-            Message nComment = new Message(sender, comment);
+            CommentMessage nComment = new CommentMessage(sender, comment);
             messages.add(nComment);
             
             
         }
         Gson gson = new Gson();
         response.setContentType("application/json;");
-        List<Message> limitedMessages = new ArrayList<>();
-        if(numberOfComments == 0){
+        List<CommentMessage> limitedMessages = new ArrayList<>();
+        if(numberOfCommentsToDisplay == 0){
             response.getWriter().println(gson.toJson(limitedMessages));
-
+            return;
         }
-        else {
-            for(int i = 0; i < numberOfComments; i++){
-                limitedMessages.add(messages.get(i));
-            }
-            response.getWriter().println(gson.toJson(limitedMessages));
-
+        for(int i = 0; i < numberOfCommentsToDisplay; i++){
```

Comment:
Nit: add one space after "for" and the opening parenthesis, and one space after the closing parenthesis.